skbio/stats/composition.py

```diff
@@ -2058,3 +2061,423 @@ def dirmult_ttest(
+
+
+def _lme_call(
+    formula,
+    metadata,
+    table,
+    groups,
+    reml=True,
+    method=None,
+    fit_kwargs={},
+    **kwargs,
+):
+    # TODO: add docs when this implementation is approved
+    # list_of_vars is essentially the list of covariates
+
+    submodels = []
+    metadata = _type_cast_to_float(metadata.copy())
+
+    data = pd.merge(table, metadata, left_index=True, right_index=True)
+    if len(data) == 0:
+        raise ValueError(
+            (
+                "No more samples left. Check to make sure that "
+                "the sample names between `metadata` and `data` "
+                "are consistent."
+            )
+        )
+
+    design_matrix = dmatrix(formula, metadata, return_type="dataframe")
+
+    # Obtaining the list of covariates by selecting the relevant columns
+    list_of_vars = design_matrix.columns.tolist()
+
+    # Removing intercept since it is not a covariate, and is included by default
+    list_of_vars.remove("Intercept")
+
+    output = []
+    for response_var in table.columns:
+        # mixed effects code is obtained here:
+        # http://stackoverflow.com/a/22439820/1167475
+        stats_formula = "%s ~ %s" % (response_var, formula)
+        model = MixedLM.from_formula(
+            formula=stats_formula, data=data, groups=groups, **kwargs
+        )
+
+        submodels.append(model)
+        results = model.fit(reml=reml, method=method, **fit_kwargs)
+
+        for var_name in list_of_vars:
+            try:
+                individial_results = {
+                    "FeatureID": response_var,
+                    "Covariate": var_name,
+                    "Log2(FC)": float(results.summary().tables[1]["Coef."][var_name]),
+                    "CI(2.5)": float(results.summary().tables[1]["[0.025"][var_name]),
+                    "CI(97.5)": float(results.summary().tables[1]["0.975]"][var_name]),
+                    "pvalue": results.pvalues[var_name],
+                }
+            except Exception:
+                try:
+                    logfc = float(results.summary().tables[1]["Coef."][var_name])
+                except Exception:
+                    logfc = np.NaN
+
+                try:
+                    ci25 = float(results.summary().tables[1]["[0.025"][var_name])
+                except Exception:
+                    ci25 = np.NaN
+
+                try:
+                    ci975 = float(results.summary().tables[1]["0.975]"][var_name])
+                except Exception:
+                    ci975 = np.NaN
+
+                individial_results = {
+                    "FeatureID": response_var,
+                    "Covariate": var_name,
+                    "Log2(FC)": logfc,
+                    "CI(2.5)": ci25,
+                    "CI(97.5)": ci975,
+                    "pvalue": results.pvalues[var_name],
+                }
+
+            output.append(individial_results)
+
+    return (output, submodels, list_of_vars)
+
+
+def dirmult_lme(
```

Comment:
There are a few opportunities to decompose blocks of code in this method, which could simplify it substantially